homeassistant/components/cloud/prefs.py

```diff
@@ -42,23 +44,64 @@
 
 STORAGE_KEY = DOMAIN
 STORAGE_VERSION = 1
-STORAGE_VERSION_MINOR = 2
+STORAGE_VERSION_MINOR = 3
 
 ALEXA_SETTINGS_VERSION = 3
 GOOGLE_SETTINGS_VERSION = 3
 
 
+class ReadOnlyStore(Store):
+    """A store which will never write."""
+
+    async def _async_handle_write_data(self, *_args: Any) -> None:
+        """Handle writing the config."""
+        # Do not write anything
+
+
+class ImportGoogleConfigStore(GoogleConfigStore):
+    """A read only configuration store for google assistant."""
+
+    # pylint: disable-next=super-init-not-called
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize a configuration store."""
+        self._hass = hass
+        self._store: Store[dict[str, Any]] = ReadOnlyStore(
+            hass,
+            self._STORAGE_VERSION,
+            self._STORAGE_KEY,
+            minor_version=self._STORAGE_VERSION_MINOR,
+        )
+
+
 class CloudPreferencesStore(Store):
     """Store cloud preferences."""
 
     async def _async_migrate_func(
         self, old_major_version: int, old_minor_version: int, old_data: dict[str, Any]
     ) -> dict[str, Any]:
         """Migrate to the new version."""
+
+        async def google_connected() -> bool:
+            """Return True if connected to Google."""
+            # If we don't have a user, we can't be connected to Google
+            if not (cur_username := old_data.get(PREF_USERNAME)):
+                return False
+
+            google_store = ImportGoogleConfigStore(self.hass)
```

Comment:
Do we not need to use the store again? Is the state of the store only changed on initialization?

Reply:
Before version this PR, the `gogle_assistant` store was shared between the cloud integration and manually configured Google assistant. With this PR, all cloud settings are stored in the `cloud` integration's store. At the first startup, settings are migrated from the `google_assistant` store to the `cloud` store. The point of the `ImportGoogleConfigStore` is to make sure we only read but do not trigger writes. I tried to make it clearer in comments.